### DISCUSSION TITLE
Optimize structure value normalization

### DISCUSSION THREAD

**Discussion by trietmnj** [USACE/go-consequences#81] [Ideas]
https://github.com/USACE/go-consequences/blob/a9f63830065e5150dd88ca01ac554c56607ec0ef/structures/structure.go#L113

This whole calculation is repeated for every frequency, but the value would be the same every time for the same structure. Could be a good place to optimize.

**Comment by HenryGeorgist** [comment]
we need to consider all uses of go-consequences. Non flood uses like earthquakes which need full valuation, and non frequency based usages. 
sometimes sacraficing optimization for consistency in approach across a diverse set of stake holders is warrented.

**Comment by trietmnj** [comment]
what's the driver for earthquake damages? is normalization based on the number of floors on a structure as a proxy for its height a good assumption?

**Comment by HenryGeorgist** [comment]
it is the only assumption i have right now. once we get the building heights algorithm for the NSI using the 3DEP data, i will be able to drive that with remote sensed data and it will be more reliable.

Shakemap is a driver for earthquake hazards, the drivers for damage are wavelengths and magnitudes and some other parameters, i havent writtten that all yet.

**Comment by trietmnj** [comment]
The two parts of that question are tied. So the driver (x-axis) of inundation and wave damages is physical depth, of which the domain is compatible with the height of that structure. If we're looking at a different multivariate driver with incompatible domains (x axis limits), then wouldn't this normalization break down?

**Reply by HenryGeorgist** [reply]
related to #86 and #85

**Comment by HenryGeorgist** [comment]
Yes, it will. But no other multivariate driver functions exist for flood at this point. The closest is erosion.

the real problem is that damage functions need to span the height of a structure, but that is not how the egm curves work.

**Comment by trietmnj** [comment]
This discussion is probably more suited for when we have a comprehensive skeleton of the system and people start to complain that it's running too slowly.

**Comment by HenryGeorgist** [comment]
agreed
